snoop/data/management/commands/importtags.py

```diff
@@ -0,0 +1,72 @@
+"""Command to import tags from a file."""
+import csv
+import logging
+
+from django.core.management.base import BaseCommand
+from snoop.data.logs import logging_for_management_command
+
+from ... import collections, models
+
+log = logging.getLogger(__name__)
+
+
+def clean_tag(tag):
+    """Remove whitespace from imported tag"""
+    tag_stripped = tag.strip()
+    cleaned_tag = tag_stripped.translate(str.maketrans({' ': '_', '\n': '_', '\t': '_', '\r': '_'}))
+    return cleaned_tag
+
+
+def read_csv(file_path):
+    """Read a csv containing md5 hashes and tags to import.
+
+    Returns: dictionary with md5 hashes as keys and a list of tags as values.
+    """
+    taglist = {}
+    with open(file_path, mode='r') as csv_file:
+        csv_reader = csv.DictReader(csv_file, fieldnames=['md5', 'tags'])
+        next(csv_reader, None)  # skip the headers
+        for row in csv_reader:
+            clean_tags = row['tags'].split(',')
+            clean_tags = [clean_tag(tag) for tag in clean_tags]
+            taglist[row['md5']] = clean_tags
+    return taglist
+
+
+def update_tags(md5, tags, collection, uuid, username, public=False):
+    """Create document tags for all new tags from the list."""
+    collection = collections.ALL[collection]
+    updated = False
+    with collection.set_current():
+        blob = models.Blob.objects.get(md5=md5)
+        digest = models.Digest.objects.get(blob=blob.pk)
+        existing_tags = models.DocumentUserTag.objects.filter(digest=digest.pk).values_list('tag', flat=True)
+        for tag in tags:
+            if tag not in existing_tags:
+                updated = True
+                models.DocumentUserTag.objects.create(digest=digest, uuid=uuid, tag=tag,
```

Comment:
use `get_or_create`, it returns a `object, created?` tuple so you can return that value.

Note: as it is, code is wrong, because the `existing_tags` query is not filtered by uuid and username.  This means you won't add some tags for some user, if another user has set the same tags.